util/util.go

```diff
@@ -58,3 +63,50 @@ func AddImageNames(store storage.Store, image *storage.Image, addNames []string)
 	}
 	return nil
 }
+
+func ContextFromGlobalOptions(c *cli.Context, flagPrefix string) (*types.SystemContext, error) {
+	ctx := &types.SystemContext{
+		RegistriesDirPath: c.GlobalString("registries.d"),
+		DockerCertPath:    c.String(flagPrefix + "cert-dir"),
+		// DEPRECATED: keep this here for backward compatibility, but override
+		// them if per subcommand flags are provided (see below).
+		DockerInsecureSkipTLSVerify: !c.GlobalBoolT("tls-verify"),
+		OSTreeTmpDirPath:            c.String(flagPrefix + "ostree-tmp-dir"),
+	}
+	if c.IsSet(flagPrefix + "tls-verify") {
+		ctx.DockerInsecureSkipTLSVerify = !c.BoolT(flagPrefix + "tls-verify")
+	}
+	if c.IsSet(flagPrefix + "creds") {
+		var err error
+		ctx.DockerAuthConfig, err = getDockerAuth(c.String(flagPrefix + "creds"))
+		if err != nil {
+			return nil, err
+		}
+	}
+	return ctx, nil
+}
+
+func parseCreds(creds string) (string, string, error) {
+	if creds == "" {
+		return "", "", errors.Wrapf(syscall.EINVAL, "credentials can't be empty")
+	}
+	up := strings.SplitN(creds, ":", 2)
+	if len(up) == 1 {
+		return up[0], "", nil
+	}
+	if up[0] == "" {
+		return "", "", errors.Wrapf(syscall.EINVAL, "username can't be empty")
+	}
```

Comment:
I think it's ok as is, or am I missing what you were concerned about?

[root@localhost buildah]# buildah from docker.io/tomsweeneyredhat/testing:first --creds=""
ERRO[0000] authentication failed for image [credentials can't be empty: invalid argument] 
[root@localhost buildah]# buildah from docker.io/tomsweeneyredhat/testing:first --creds=
ERRO[0000] authentication failed for image [credentials can't be empty: invalid argument] 
[root@localhost buildah]# buildah from docker.io/tomsweeneyredhat/testing:first 
ERRO[0000] error pulling image "docker.io/tomsweeneyredhat/testing:first": Error initializing image from source docker://tomsweeneyredhat/testing:first: errors:
denied: requested access to the resource is denied
unauthorized: authentication required